skopt/utils.py

```diff
@@ -441,8 +442,7 @@ def point_aslist(search_space, point_as_dict):
 
 
 def normalize_dimensions(dimensions):
```

Comment:
changes here are to help `normalize_dimensions` to also work when you pass a list of `Dimension` objects or a `Space`. Felt like this was unclear/complex before because `check_dimension` would simply return the value you pass unchanged if it was already a `Dimension` instance.